x/stakeibc/keeper/unbonding.go

```diff
@@ -225,12 +246,17 @@ func SortUnbondingCapacityByPriority(validatorUnbondCapacity []ValidatorUnbondCa
 // Returns the list of messages and the callback data for the ICA
 func (k Keeper) GetUnbondingICAMessages(
 	hostZone types.HostZone,
-	totalUnbondAmount sdkmath.Int,
+	totalNativeAmount sdkmath.Int,
+	totalStAmount sdkmath.Int,
 	prioritizedUnbondCapacity []ValidatorUnbondCapacity,
 	batchSize int,
-) (msgs []proto.Message, unbondings []*types.SplitDelegation, err error) {
+) (msgs []proto.Message, unbondings []*types.SplitUndelegation, err error) {
+	// Calculate the implied redmeption rate from the totals
+	impliedRedemptionRate := sdk.NewDecFromInt(totalNativeAmount).Quo(sdk.NewDecFromInt(totalStAmount))
```

Comment:
I didn't follow that last sentence completely but it feels important.  Can you rephrase it or go into more detail?  It would seem that the stToken and nativeAmounts must always hit zero at the same time no?

Reply:
I responded to this in the other PR as well, but here's an example:

```
EUR 1 - RR: 1.0, Native: 1000, StTokens: 1000
EUR 2 - RR: 1.5, Native: 1500, StTokens: 1000
EUR 3 - RR: 2.0, Native: 2000, StTokens: 1000
```

When we submit the unbonding, we submit it for the total which is 4500 native, and 3000 sttokens.
This has an implied rate of 4500/3000 = 1.5

However, let's say the first batch only unbonds 2500 tokens.
Since we calculate the number of stTokens using the implied rate of 1.5, this means we store 1666 sttokens on that batch

So when we decrement those amounts from both records we end up with:
```
EUR 1 - Native: 0, StTokens: 0            // subtracts 1000n, 1000st and "zero's" them both out 
EUR 2 - Native: 0, StTokens: 334          // subtracts the remaining 1500n, and 666st - only the native tokens are zero'd out
EUR 3 - Native: 2000, StTokens: 1000
```

So my point was that they wont zero out at the same time, but once all the other batches completes, then the last record will zero out together

Reply:
So I think the other way to solve this is to just compute the number of stTokens in the callback using the ratio in the EUR. The only issue is then you get some precision error on the last validator so you probably need a clause that checks if the stToken amount is smaller than some epsilon, then round it down to 0

Reply:
hmm, yeah actually maybe it does work! cause you're right in that it would just use the totals next epoch, so the zero doesn't really make a difference

Reply:
This morning we discussed a different approach to re-compute the number of stTokens in the callback based on the RR from the individual record. Assuming both work, do you have a preference?

Approach #1:
* only does the stToken calculation once per epoch
* handles precision error gracefully
* but, has this weird edge case above during retries where the amounts don't match

Approach #2:
* does stToken calculation each callback
* handling precision error is more hand wavy
* but, amounts always match the record's RR

cc @ethan-stride

Reply:
I don't have a strong preference but lean towards #2 to prioritize having amounts that are interpretable and diagnosable. As long as we are confident the precision errors will not compound and won't cause errors that halt the system. Wby?

Reply:
Might be good to get additional input from @shellvish @asalzmann on this one.

Reply:
discussed offline - moving forward with approach #2 from above for now and will revisit refreshing the RR again during retries later